### DISCUSSION TITLE
Feature Request (GitHub Actions): Access control for workflow logs in public repositories

### DISCUSSION THREAD

**Discussion by stekern** [community/community#64813] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently every authenticated GitHub user has access to GitHub Actions workflow logs in public repositories. In many cases, however, you may have public repositories where you only want the owner of the repository or users with explicit access to the repository to have access to these logs.

In accordance with the principle of least privilege, it would be very useful if GitHub introduced finer controls over workflow log access in public repositories as this would especially mitigate the risks associated with accidentally exposing sensitive information.

Are there any plans for implementing something like this?

**Comment by eliheady** [comment]
This.

I have another example: I am working on a private fork of a public repo and I want to make logs from arbitrary private workflow runs public for a feature I am working on.

**Comment by bepoooe** [comment]
I understand your concern about the current access model for GitHub Actions workflow logs in public repositories. You're highlighting an important security principle - the principle of least privilege - where users should only have access to the minimum information needed for their role.
Currently, as you correctly noted, all authenticated GitHub users can view workflow logs in public repositories, which could potentially expose sensitive information if accidentally included in these logs.
Regarding your question about plans for implementing finer controls over workflow log access in public repositories - I don't have specific information about GitHub's product roadmap beyond my knowledge cutoff of October 2024. I can't confirm whether this feature has been announced or implemented since then.
However, there are some workarounds you might consider in the meantime:

Use repository secrets for sensitive information to prevent it from appearing in logs
Consider using private repositories for workflows containing sensitive information
Be vigilant about what gets logged in your workflows
Use environment protection rules for deployments with sensitive information

I'd recommend raising this as a feature request through GitHub's official feedback channels if you haven't already, as it does seem like a reasonable security enhancement that would benefit many users.
